### PR TITLE
[lro] polling strategy not generate the line when serviceVersion and endpoint is null

### PR DESCRIPTION
**Pull request by haolingdong-msft**
follow up pr on https://github.com/Azure/autorest.java/pull/2349#discussion_r1371344306

### CODE REVIEW

File: javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java [hunk 1]
```diff
@@ -1504,60 +1504,22 @@ private String getPagingNextPageExpression(ClientMethod clientMethod, String met
...
         return clientMethod.getMethodPollingDetails().getSyncPollingStrategy()
                 .replace("{httpPipeline}", clientMethod.getClientReference() + ".getHttpPipeline()")
-                .replace("{endpoint}", endpoint)
+                .replace("    .setEndpoint({endpoint}))", getEndpointPollingStrategy(clientMethod))
                 .replace("{context}", contextParam)
-                .replace("{serviceVersion}", getServiceVersionValue(clientMethod))
+                .replace("    .setServiceVersion({serviceVersion}))", getServiceVersionPollingStrategy(clientMethod))
                 .replace("{serializerAdapter}", clientMethod.getClientReference() + ".getSerializerAdapter()")
                 .replace("{intermediate-type}", clientMethod.getMethodPollingDetails().getIntermediateType().toString())
                 .replace("{final-type}", clientMethod.getMethodPollingDetails().getFinalType().toString());
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java:1525]: Why this complexity? I kind of assume we only need to add one line?

```java
        return clientMethod.getMethodPollingDetails().getSyncPollingStrategy()
                .replace("{httpPipeline}", clientMethod.getClientReference() + ".getHttpPipeline()")
                .replace("{endpoint}", endpoint)
                .replace("{context}", contextParam)
                .replace("{serviceVersion}", getServiceVersionValue(clientMethod))
                .replace("{serializerAdapter}", clientMethod.getClientReference() + ".getSerializerAdapter()")
                .replace("{intermediate-type}", clientMethod.getMethodPollingDetails().getIntermediateType().toString())
                .replace("{final-type}", clientMethod.getMethodPollingDetails().getFinalType().toString());
                .replace(".setServiceVersion(null)", "");
```
last line

Comment (haolingdong-msft) [reply javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java:1525]: Ah, this is better. I will update and test. Maybe `.replace("    .setServiceVersion(null)", "")` to remove the front spaces/tab as well, also`endpoint` is in the middle, and may need to handle the space and `\n`, I'm currently testing.

Comment (weidongxu-microsoft) [reply javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java:1525]: Don't put the space there. Formatter should be able to handle that.

Comment (haolingdong-msft) [reply javagen/src/main/java/com/azure/autorest/template/ClientMethodTemplate.java:1525]: ok, I will test.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**